### PR TITLE
Fix missing commit hash if app started directly

### PR DESCRIPTION
**Pull request by Cruikshanks**
If the service is run on Heroku than we have no issue with the commit hash. We have added a buildpack that reads the latest git commit during build, and then creates an env var called `GIT_SHA` with the hash as its value.

Locally if you start the app using gulp it also resolves the issue. There is a gulp task which is called when the default task is run, and it uses a gulp addon to obtain the value, and then also creates an env var called `GIT_SHA`.

The problem is if you don't use gulp to start the app, for example `node server.js` or as it came to light for me `pm2 start server.js`, you never get an env var called `GIT_SHA` created.

This means on the version and healthcheck page the commit hash displays as `unknown`. This fix is to implement a solution for when the app is not called from either Heroku or gulp.

### CODE REVIEW

File: src/services/commitHash.service.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    // At this point we accept we don't have it and go with Unknown
+    if (!commitReference) {
+      commitReference = 'Unknown'
+    }
+
+    // Whatever the result by now, we set the env var to the value we have. This
+    // means after the first call to commitHash() subsequent calls should return
+    // immediately after reading GIT_SHAs
+    process.env['GIT_SHA'] = commitReference
```
Comment (irisfaraway) [inline src/services/commitHash.service.js]: May just have to do with setting the value as opposed to reading it, but is there a reason it's `process.env['GIT_SHA']` here and `process.env.GIT_SHA` further up?

File: src/services/commitHash.service.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    // If the env var is set immediately return the value we read from it here
+    let commitReference = process.env.GIT_SHA
+    if (commitReference) {
+      return commitReference
+    }
+
+    // If the env var was not set then we next prioritise reading from a file
+    // called GIT_SHA. This is to fit in with our deployment process
+    // (non-heroku) which will create this file and add the current commit hash
+    // to it
```
Comment (irisfaraway) [inline src/services/commitHash.service.js]: Are we creating this file yet or will that be a separate piece of work? Just curious to see what it looks like.

Comment (Cruikshanks) [reply src/services/commitHash.service.js]: Separate piece of work. Its part of the deployment process I've been beavering away at, which you can find in our internal GitLab repo if you are curious.

File: src/services/commitHash.service.js (hunk 1 above)
Comment (Cruikshanks) [reply src/services/commitHash.service.js]: 🤔 

No idea, probably based on where I copied and pasted it from 😳 ! Will give see if I can use the same syntax for both.

**Pull request merged**